### PR TITLE
[feat][feat] Add FairLaunchRelayer contract 

### PR DESCRIPTION
**Pull request by alpanaca**
## Description
A contract that will claim alpaca from FairLaunch contract and send it to MiniFL on Fantom via Anyswap

**MERGE AFTER #237** 

## Why?
Since we have launched LYF on Fantom. We need to continuously bridge Alpaca token from BSC to Fantom so that the MiniFL contract will have enough token to give out users.

## ChangeLogs:
- new FairLaunchRelayer contract
- Mock contracts
- Unit test on FairLaunchRelayer contract

### CODE REVIEW

File: contracts/8.10/protocol/FairLaunchRelayer.sol [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    (bool _success, ) = address(fairLaunch).call(abi.encodeWithSelector(0xddc63262, fairLaunchPoolId));
+    if (_success) emit LogFairLaunchHarvest(address(this), token.myBalance() - _before);
+  }
+
+  /// @notice Harvest reward from FairLaunch and send it to another chain destination address
+  function forwardToken() external {
+    _fairLaunchHarvest();
+    uint256 _forwardAmount = token.myBalance();
+    // If the amount is too small, the cross chain fee won't be plausible
+    if (_forwardAmount < (1000 * 1e18)) {
```
Comment (alpanaca) [inline contracts/8.10/protocol/FairLaunchRelayer.sol:125]: Do we need upper bound check?

File: contracts/8.10/protocol/FairLaunchRelayer.sol [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  event LogFairLaunchWithdraw();
+  event LogFairLaunchHarvest(address _caller, uint256 _harvestAmount);
+  event LogForwardToken(address _destination, uint256 _forwardAmount);
+
+  /// @notice Errors
+  error FairLaunchRelayer_StakeTokenMismatch();
+  error FairLaunchRelayer_AmoutTooSmall();
+  error FairLaunchRelayer_AlreadyDeposited();
+
+  /// @notice State
```
Comment (spicysquid168) [inline contracts/8.10/protocol/FairLaunchRelayer.sol:42]: Add `name` too

File: contracts/8.10/protocol/FairLaunchRelayer.sol (hunk 1 above)
Comment (spicysquid168) [reply contracts/8.10/protocol/FairLaunchRelayer.sol:125]: No need

File: contracts/8.10/protocol/FairLaunchRelayer.sol [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    uint256 _before = token.myBalance();
+    // Low level call: this shouldn't block forward token even if failed to harvest
+    (bool _success, ) = address(fairLaunch).call(abi.encodeWithSelector(0xddc63262, fairLaunchPoolId));
+    if (_success) emit LogFairLaunchHarvest(address(this), token.myBalance() - _before);
+  }
+
+  /// @notice Harvest reward from FairLaunch and send it to another chain destination address
+  function forwardToken() external {
+    _fairLaunchHarvest();
+    uint256 _forwardAmount = token.myBalance();
```
Comment (spicysquid168) [inline contracts/8.10/protocol/FairLaunchRelayer.sol:123]: `token.myBalance() >= maxCrossChainAmount ? maxCrossChainAmount : token.myBalance()`

**Comment by spicysquid168** [comment]
@alpanaca can you change merge it to `main` instead of to `feat/delta-neutral`?

**Comment by alpanaca** [comment]
Can do but the will contain the change of delta neutral branch. Can we merge #237 first?

**Comment by spicysquid168** [comment]
After I deploy mainnet

**Comment by alpanaca** [comment]
Ok. Will change the merge to main after that

File: contracts/8.10/protocol/FairLaunchRelayer.sol (hunk 3 above)
Comment (alpanaca) [reply contracts/8.10/protocol/FairLaunchRelayer.sol:123]: should the maxCrossChainAmount be hardcoded. Or state variable with setter for later changes?

Comment (spicysquid168) [reply contracts/8.10/protocol/FairLaunchRelayer.sol:123]: state variable with setter _(with Axolo)_

**Comment by alpanaca** [comment]
Hmm..Axolo won't catch commit pushed

File: test/integrations/protocol/FairLaunchRelayer.test.ts [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+
+/// Contracts
+let relayer: FairLaunchRelayer;
+let relayerAsAlice: FairLaunchRelayer;
+
+describe("FairLaunchRelayer", () => {
+  async function fixture() {
+    [deployer, alice] = await ethers.getSigners();
+
+    // Mint
```
Comment (alpanaca) [inline test/integrations/protocol/FairLaunchRelayer.test.ts:50]:
```suggestion
    // Deploy ERC-20
```

**Pull request merged**